### PR TITLE
Add dem_stitcher recipe

### PR DESCRIPTION
**Pull request by cmarshak**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jhkennedy** [comment]
I am willing to be a maintainer

**Comment by jhkennedy** [comment]
@conda-forge-admin, please lint

**Comment by jhkennedy** [comment]
@conda-forge-admin, please lint

**Comment by cmarshak** [comment]
@conda-forge/help-python, ready for review!

File: recipes/dem_stitcher/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  sha256: e211e94e115e9c84fef7e2fb11f75b56076c940f31d04a2422be60c9db27e24c
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7,<3.11
```
Comment (jhkennedy) [inline recipes/dem_stitcher/meta.yaml:19]:
```suggestion
    - python >=3.7
```

File: recipes/dem_stitcher/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - pip
+    - setuptools >=42
+    - setuptools_scm >=3.4
+    - wheel
+  run:
+    - python >=3.7,<3.11
```
Comment (jhkennedy) [inline recipes/dem_stitcher/meta.yaml:25]:
```suggestion
    - python >=3.7
```

**Review by jhkennedy — CHANGES_REQUESTED** [review]
I think in general we should avoid a top pin if there's no known issues.

**Review by jhkennedy — APPROVED** [review]

**Comment by jhkennedy** [comment]
@conda-forge/staged-recipess and or @conda-forge/help-python this should be ready to merge!

**Pull request merged**